### PR TITLE
Iperf not executing

### PR DESCRIPTION
**Pull request by TasmiyaNalatwad**
NIC : iperf local variable 'tput' referenced before assignment

Signed-off-by: Tasmiya Nalatwad <tasmiya@linux.vnet.ibm.com>

[root@ltcrain41-lp4 net]# avocado run --test-runner runner iperf_test.py -m iperf_test.py.data/iperf_virt_test.yaml
JOB ID     : 0143a7cff6751bbacd0d5b58105327bdfd551e26
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-04-06T08.34-0143a7c/job.log
 (1/2) iperf_test.py:Iperf.test;run-hbond-mtu-1500-0b8a: PASS (64.69 s)
 (2/2) iperf_test.py:Iperf.test;run-hbond-mtu-9000-96d2: PASS (76.30 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-04-06T08.34-0143a7c/results.html
JOB TIME   : 142.34 s

Attaching job.log
[iperf.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/8427591/iperf.log)

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
@TasmiyaNalatwad I feel we should have library in utils to start/stop firewall services. please check on this.
if not I feel better we add in utils as many network/other test cases will be needing this.
@abdhaleegit @vaishnavibhat ^^

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@TasmiyaNalatwad there is already a  function `SpecificServiceManager` in service utils , please make use of this for firewall service restart

File: io/net/iperf_test.py [hunk 1]
```diff
@@ -80,6 +80,17 @@ def setUp(self):
             if not output.exit_status == 0:
                 self.cancel("unable to install the package %s on peer machine "
                             % pkg)
+        cmd1 = "systemctl stop firewalld.service"
+        cmd2 = "systemctl status firewalld.service"
```
Comment (abdhaleegit) [inline io/net/iperf_test.py:84]: The same command may not work for SUSE os ?  take care of that

File: io/net/iperf_test.py [hunk 2]
```diff
@@ -194,6 +207,17 @@ def tearDown(self):
         """
         Killing Iperf process in peer machine
         """
+        cmd1 = "systemctl start firewalld.service"
+        cmd2 = "systemctl status firewalld.service"
```
Comment (abdhaleegit) [inline io/net/iperf_test.py:211]: duplicate commands in setup and cleanup

**Review by abdhaleegit — COMMENTED** [review]
Refer to avocado/utils/multipath.py  code to run service on different distros

File: io/net/iperf_test.py [hunk 3]
```diff
@@ -100,18 +100,39 @@ def setUp(self):
...
+        elif detected_distro.name == "centos":
+            cmd_fw = "service iptables stop"
+        else:
+            self.cancel("Distro not supported")
+        if process.system(cmd_fw, ignore_status=True, shell=True) != 0:
+            self.cancel("Unable to disable firewall")
+        output = self.session.cmd(cmd_fw)
+        if not output.exit_status == 0:
+            self.cancel("Unable to disable firewall on peer")
+
```
Comment (vaishnavibhat) [inline io/net/iperf_test.py:121]: LGTM

**Comment by Naresh-ibm** [comment]
@abdhaleegit @TasmiyaNalatwad @vaishnavibhat 
So, we are not pushing this to common utils?
@TasmiyaNalatwad I do not see disabling firewall on host side.

**Comment by TasmiyaNalatwad** [comment]
Hi @Naresh-ibm  We already have common function in utils for disabling firewall , but that works for only host setup. Using that common function we cannot disable firewall on peer lpar where peer is connected via ssh.

using common function for host and writing one extra function for peer in the iperf_test.py file to disable firewall was like redundant lines. To avoid redundant lines usage wrote only one handle to disable firewall on both host and peer.

The condition is for disabling firewall on both host and peer.

@vaishnavibhat @abdhaleegit  Requesting your inputs

**Review by PraveenPenguin — APPROVED** [review]

File: io/net/iperf_test.py (hunk 2 above)
Comment (TasmiyaNalatwad) [reply io/net/iperf_test.py]: not using the duplicate cmds, modified as suggested.

File: io/net/iperf_test.py (hunk 1 above)
Comment (TasmiyaNalatwad) [reply io/net/iperf_test.py]: Checking for Distro and OS versions and making use of commands

**Pull request merged**